### PR TITLE
Add minimal tests to libvirt and streamline defaults

### PR DESCRIPTION
**Pull request by 14rcole**
Minimal tests for a given provider verify that each role in a provider will run successfully with only the attributes defined as 'required' in the schema.  There are currently problems with default values not being set properly and causing provisioning to fail when optional attributes are not provided.

This also separates default values for libvirt to a 'defaults.yml' file which is used to populate res_defs at the beginning of provisioning.  All providers will eventually move to this method in order to minimize the repeated usage of ansible's 'default()' filter which, while useful, can lead to the problems described above.

### CODE REVIEW

**Comment by 14rcole** [comment]
@JunqiZhang0 you're working on adding azure support, right?  You may want to take a look at the changes I've made to the libvirt module here regarding defaults.  I think that all modules should follow this pattern from here on out

**Comment by samvarankashyap** [comment]
@14rcole The reason why it failed on privileged psi is the following fix has to be merged
#1286

**Comment by JunqiZhang0** [comment]
@samvarankashyap Thanks

File: linchpin/provision/roles/libvirt/tasks/provision_res_defs.yml [hunk 1]
```diff
@@ -1,13 +1,34 @@
...
-  set_fact:
-    uri_hostname: "{{ res_def['uri'] | get_host_from_uri }}"
+
+- name: "Parse default variable file"
+  template:
+    src: "../files/defaults.yml"
+    dest: "/tmp/linchpin-defaults.yml"
+
+- name: "Load default variables"
+  include_vars: '/tmp/linchpin-defaults.yml'
```
Comment (abraverm) [inline linchpin/provision/roles/libvirt/tasks/provision_res_defs.yml]: random filename and remove it later

**Comment by samvarankashyap** [comment]
@14rcole Did you rebase with recent merges?

**Comment by samvarankashyap** [comment]
@14rcole please rebase

**Comment by 14rcole** [comment]
[test]

**Comment by JunqiZhang0** [comment]
[test]

**Comment by 14rcole** [comment]
@samvarankashyap did you get libvirt tests working?

**Comment by samvarankashyap** [comment]
@14rcole Nope I have not, I am current working on bug fixes that need to happen before enabling libvirt. similar to
#1263

**Comment by samvarankashyap** [comment]
@14rcole could you resolve the conflicts and remove the debug statements.

**Comment by 14rcole** [comment]
[test]

Tests seem to have failed because of an unrelated aws issue

File: linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml [hunk 2]
```diff
@@ -208,8 +208,9 @@
   with_nested:
     - ["{{ libvirt_resource_name }}"]
     - "{{ res_count }}"
-    - ["{{ res_def['name_separator'] | default('_')  }}"]
-    - ["{{ res_def['domain'] | default('example.com') }}"]
+<<<<<<< HEAD
```
Comment (samvarankashyap) [inline linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml]: @14rcole I think you have to delete this.

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by samvarankashyap** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[merge]

**Pull request merged**